app/src/main/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonsLayout.kt

```diff
@@ -33,6 +35,12 @@ internal fun LessonsLayout(viewModel: LessonsViewModel = viewModel(), onEvent: (
     val lessons by viewModel.lessons.collectAsState(emptyList())
     val selectedLanguage by viewModel.selectedLanguage.collectAsState()
 
+    val appLanguage by viewModel.settings.appLanguageFlow.collectAsState()
+
+    LaunchedEffect(appLanguage) {
+        viewModel.updateSelectedLanguage(Language(appLanguage))
+    }
```

Comment:
You are putting business logic in the view layer, can you rework this so that this business logic is in the ViewModel

Reply:
And can you write unit tests to test this behavior